gamla/optimized/sync.py

```diff
@@ -421,3 +421,28 @@ def bifurcate(*funcs):
     (15, 5)
     """
     return compose_left(iter, lambda it: itertools.tee(it, len(funcs)), stack(funcs))
+
+
+def flatten(iterable: Iterable[Union[Any, Iterable]]) -> Iterable[Any]:
+    """Flatten a given iterable revursively.
+    >>> iter = [1, "a", frozenset({"something"}), (("hi", 6),)]
+    >>> flatten(iter)
+    (
+        1,
+        "a",
+        "something",
+        "hi",
+        6,
+    )
+    """
+    should_be_concated = anyjuxt(*map(operator.is_instance)([frozenset, tuple, list]))
```

Comment:
how about using gamla.all_juxt(complement(gamla.is_instance)(str), operator.is_iterable)
also consider moving this outside so we won't need to run it every invocation

Reply:
I thought about this, but like in spacy there can be dataclasses that our iterables, so I prefer to give it explicitly what we want to conact